client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -29,16 +51,87 @@ const getIframeUrl = ( siteAdminUrl, postId, postType ) => {
 };
 
 class CalypsoifyIframe extends Component {
+	state = {
+		isMediaModalVisible: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.iframe = null;
+	}
+
+	componentDidMount() {
+		window.addEventListener( 'message', this.onMessage, false );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'message', this.onMessage, false );
+	}
+
+	setIframeRef = element => {
+		this.iframe = element;
+	};
+
+	onMessage = ( { data } ) => {
+		if ( typeof data !== 'string' || data[ 0 ] !== '{' ) {
+			return;
+		}
+		const message = JSON.parse( data );
+		const { action, type, payload } = message;
+
+		if ( type !== 'gutenbergIframeMessage' ) {
+			return;
+		}
```

Comment:
we probably also want to perform some security checking here to try and make sure that the messages are originating from the iframe.

when we add `window.addEventListener( 'message', this.onMessage, false );` we're starting to listen to _all_ messages, even from the console, or from other embedded frames that could possibly exist, such as in the post preview modal.

we can verify the `message.origin` which is set by the browser and compare it against the site we've loaded

```js
// something like this
if ( ! message.origin.startsWith( this.props.siteDomain ) ) {
	return;
}
```